core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -66,7 +100,8 @@ private[spark] object TaskDescription {
     }
   }
 
-  def encode(taskDescription: TaskDescription): ByteBuffer = {
+  @throws[TaskNotSerializableException]
+  def encode(taskDescription: TaskDescription, serializedTask: ByteBuffer): ByteBuffer = {
```

Comment:
The previous version had only one parameter, and the two parameter  in  here is more readable